test/e2e/suites/e2e/e2e_test.go

```diff
@@ -129,6 +131,47 @@ var _ = Describe("e2e tests", func() {
 	})
 
 	Describe("Workload cluster (without lb)", func() {
+		It("Should create port(s) with custom options", func() {
+			shared.Byf("Creating a cluster")
+			clusterName := fmt.Sprintf("cluster-%s", namespace.Name)
+			configCluster := defaultConfigCluster(clusterName, namespace.Name)
+			configCluster.ControlPlaneMachineCount = pointer.Int64Ptr(1)
+			configCluster.WorkerMachineCount = pointer.Int64Ptr(0)
```

Comment:
`Creating MachineDeployment with custom port options` line 143 I assume this means create worker machines

but here seems the worker is 0 , is it expected or typo?

Reply:
We do not need a worker in this case.
If we can the ports being created on at least one server, then it is verifies the feature.

Reply:
ok, because I assume below line 144 (-md-3) is actually a machinset for worker machine
if we don't have that purpose to test machineset of worker node, then why we need add those code there..

Reply:
You are right. In fact, we are just verifying the port creation, not usage.
The `makeOpenStackMachineTemplateWithPortOptions` is the one that adds the feature which creates the worker's spec.
I think I should change the `WorkerMachineCoun` as you have pointed out earlier.